src/main/java/com/zendesk/maxwell/MaxwellParser.java

```diff
@@ -78,18 +80,48 @@ public void start() throws Exception {
 		this.replicator.start();
 	}
 
-	public void stop() throws Exception {
+	public void stop() {
+		// note: we use stderr in this function as LOGGER.err() oftentimes
+		// won't flush in time, and we lose the messages.
+
 		this.binlogEventListener.stop();
-		this.replicator.stop(5, TimeUnit.SECONDS);
+		try {
+			this.replicator.stop(5, TimeUnit.SECONDS);
```

Comment:
this is a bit weird. Considering there is a chance that L129 (https://github.com/zendesk/maxwell/pull/111/files#diff-5adf358ee2ff00689a314b88b7f2786cR129 ) may simply restart it in the main thread. 

I suggest setting the shouldStop flag here, and maybe letting the main thread do this cleanup?